From bac97458bf52e5bcd5588b94435e402c44106496 Mon Sep 17 00:00:00 2001
From: Fan Du <fan.du@intel.com>
Date: Thu, 26 May 2016 12:55:29 -0500
Subject: [PATCH 207/208] pci {guest kernel}: set pci net class bar to 4

For pci net class is defined as 0x020000
By optimaztion, guess_bar_count set net class bar count to 2.
This will cause subsquent bar resource missing after lkvm
commit [1] extend virtio net bar resouces with MMIO bar.

[1] a463650caad6 ("kvm tools: pci: add MMIO interface to virtio-pci devices")

Signed-off-by: Fan Du <fan.du@intel.com>

Update to enable SRIOV on CC2.0
Change suggested by Anthony Xu <anthony.xu@intel.com>

PCI device has 6 BARs, linux kernel normally scan all 6 BARs to
configure the MMIOs.  Since scanning BAR operation is expensive
in VM and virtual NIC(virtio nic) use 3 BARs at most, there is
an optimization in clear-container guest KERNEL, which only scans
the first 3 BAR for NIC devices. Unfortunately igbvf uses BAR 3,
which is the 4th BAR, this BAR is not configured due to above
reason in Clear-Container. BAR 3 in igbvf is pointing to PBA,
which is used by MSI-X, that's why guest kernel failed to
configure MSI-X interrupt for igbvf.
---
 drivers/pci/probe.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pci/probe.c b/drivers/pci/probe.c
index cbe81b100d51..8b7dbd221af3 100644
--- a/drivers/pci/probe.c
+++ b/drivers/pci/probe.c
@@ -1357,7 +1357,7 @@ static int guess_bar_count(int class)
         if (class == 0x068000)
             return 0;
         if (class == 0x020000)
-            return 2;
+            return 4;
         if (class == 0x010000)
             return 2;
         if (class == 0x00ff00)
-- 
2.11.0

